lib/book/book_page/book_page_view.dart

```diff
@@ -71,24 +71,60 @@ class _BookPageViewState extends State<BookPageView> {
                   ? Text(widget.book.title)
                   : Text(pages[currentPageIndex].chapter!.title),
               actions: [
-                IconButton(
-                    onPressed: () async {
-                      final BookPage? result =
-                          await Navigator.pushNamed<dynamic>(
-                              context, newPageRoute,
-                              arguments: <String, dynamic>{
-                            "newPage": pages[currentPageIndex],
-                            "bookId": widget.book.docId,
-                            "chapters": widget.book.bookData!.bookChapters,
-                            "pageMode": BookPageMode.edit,
-                          });
-                      if (result != null) {
-                        context
-                            .read<BookPageBloc>()
-                            .add(UpdateBookPagesEvent(page: result));
-                      }
-                    },
-                    icon: Icon(Icons.edit)),
+                pages.isNotEmpty
+                    ? IconButton(
+                        onPressed: () async {
+                          final BookPage? result =
+                              await Navigator.pushNamed<dynamic>(
+                                  context, newPageRoute,
+                                  arguments: <String, dynamic>{
+                                "newPage": pages[currentPageIndex],
+                                "bookId": widget.book.docId,
+                                "chapters": widget.book.bookData!.bookChapters,
+                                "pageMode": BookPageMode.edit,
+                              });
+                          if (result != null) {
+                            context
+                                .read<BookPageBloc>()
+                                .add(UpdateBookPagesEvent(page: result));
+                          }
+                        },
+                        icon: Icon(Icons.edit))
+                    : Container(),
+                pages.isNotEmpty
+                    ? IconButton(
+                        onPressed: () async {
+                          final result =  await showDialog(
```

Comment:
Since we use a lot of AlertDialogs in onBackPressed functions and here, you could create a static function for this AlertDialog that is almost the same in every case. If you are interested I can send you what I have already coded.